modules/acm/src/main/java/edu/gemini/epics/acm/CaParameterStorage.java

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright (c) 2016-2018 Association of Universities for Research in Astronomy, Inc. (AURA)
+ * For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+package edu.gemini.epics.acm;
+
+import edu.gemini.epics.EpicsWriter;
+import gov.aps.jca.CAException;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+class CaParameterStorage {
```

Comment:
What is the advantage of separating out `CaParameterStorage` from `CaCommandSenderImpl`?

It seems now like `CaCommandSenderImpl` is possibly just an unnecessary layer of abstraction as an interface to this class, or am I missing something?

Is this so that you can use it in `CaTaskControlImpl` as well? Something feels a bit repetitive here, but maybe I just haven't seen enough of the PR yet (I'm commenting as I go).

Reply:
I separate it to be able to use it in `CaTaskRecord`, the other type of record that has parameters.